`rocmlir-rock` failure seems unrelated:
```
FAILED: mlir/lib/Dialect/Rock/Transforms/CMakeFiles/obj.MLIRRockTransforms.dir/ViewToTransform.cpp.o
/nix/store/k931312cnlqknjnnpixwjhysq78ij7dx-rocm-llvm-clang-wrapper-5.7.1/bin/clang++ -DGTEST_HAS_RTTI=0 -D_DEBUG -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/build/source/build/mlir/lib/Dialect/Rock/Transforms -I/build/source/>
In file included from /build/source/mlir/lib/Dialect/Rock/Transforms/ViewToTransform.cpp:14:
/build/source/mlir/include/mlir/Conversion/TosaToRock/TosaToRock.h:21:10: fatal error: 'mlir/Conversion/RocMLIRPasses.h.inc' file not found
#include "mlir/Conversion/RocMLIRPasses.h.inc"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

--- a/pkgs/development/python-modules/onnx/default.nix
+++ b/pkgs/development/python-modules/onnx/default.nix
@@ -1,24 +1,30 @@
 {
   lib,
-  stdenv,
   buildPythonPackage,
-  cmake,
   fetchFromGitHub,
-  gtest,
-  nbval,
-  numpy,
-  parameterized,
-  protobuf_21,
+
+  # build-system
+  cmake,
   pybind11,
-  pytestCheckHook,
-  pythonOlder,
-  tabulate,
-  typing-extensions,
-  abseil-cpp,
-  google-re2,
-  pillow,
-  protobuf,
   setuptools,
+
+  # nativeBuildInputs
+  protobuf-core,
+
+  # buildInputs
+  abseil-cpp,
+  protobuf,
+  gtest,
+
+  # dependencies
+  numpy,
+
+  google-re2,
+  nbval,
+  parameterized,
+  pillow,
+  pytestCheckHook,
+  tabulate,
 }:
 
 let
@@ -28,8 +34,6 @@ buildPythonPackage rec {
   pname = "onnx";
   version = "1.17.0";
   pyproject = true;
-
-  disabled = pythonOlder "3.8";
 
   src = fetchFromGitHub {
     owner = "onnx";
@@ -45,12 +49,11 @@ buildPythonPackage rec {
   ];
 
   nativeBuildInputs = [
-    protobuf_21 # for protoc
+    protobuf-core # `protoc` required
   ];
 
   buildInputs = [
     abseil-cpp
-    protobuf_21
     gtestStatic
     pybind11
   ];
@@ -58,7 +61,6 @@ buildPythonPackage rec {
   dependencies = [
     protobuf
     numpy
-    typing-extensions
   ];
 
   nativeCheckInputs = [
@@ -109,31 +111,6 @@ buildPythonPackage rec {
     "examples"
   ];
 
-  disabledTests =
-    [
-      # attempts to fetch data from web
-      "test_bvlc_alexnet_cpu"
-      "test_densenet121_cpu"
-      "test_inception_v1_cpu"
-      "test_inception_v2_cpu"
-      "test_resnet50_cpu"
-      "test_shufflenet_cpu"
-      "test_squeezenet_cpu"
-      "test_vgg19_cpu"
-      "test_zfnet512_cpu"
-    ]
-    ++ lib.optionals stdenv.hostPlatform.isAarch64 [
-      # AssertionError: Output 0 of test 0 in folder
-      "test__pytorch_converted_Conv2d_depthwise_padded"
-      "test__pytorch_converted_Conv2d_dilated"
-      "test_dft"
-      "test_dft_axis"
-      # AssertionError: Mismatch in test 'test_Conv2d_depthwise_padded'
-      "test_xor_bcast4v4d"
-      # AssertionError: assert 1 == 0
-      "test_ops_tested"
-    ];
-
   __darwinAllowLocalNetworking = true;
 
   postCheck = ''
@@ -143,11 +120,11 @@ buildPythonPackage rec {
 
   pythonImportsCheck = [ "onnx" ];
 
-  meta = with lib; {
-    changelog = "https://github.com/onnx/onnx/releases/tag/${lib.removePrefix "refs/tags/" src.rev}";
+  meta = {
     description = "Open Neural Network Exchange";
     homepage = "https://onnx.ai";
-    license = licenses.asl20;
-    maintainers = with maintainers; [ acairncross ];
+    changelog = "https://github.com/onnx/onnx/releases/tag/${lib.removePrefix "refs/tags/" src.rev}";
+    license = lib.licenses.asl20;
+    maintainers = with lib.maintainers; [ acairncross ];
   };
 }

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -9282,7 +9282,9 @@ self: super: with self; {
 
   onlykey-solo-python = callPackage ../development/python-modules/onlykey-solo-python { };
 
-  onnx = callPackage ../development/python-modules/onnx { };
+  onnx = callPackage ../development/python-modules/onnx {
+    protobuf-core = pkgs.protobuf;
+  };
 
   onnxconverter-common = callPackage ../development/python-modules/onnxconverter-common {
     inherit (pkgs) protobuf;

